csrc/multidevice/utils.h

```diff
@@ -86,8 +86,15 @@ bool haveDifferentShardings(
 // Returns whether a resharding expr reshards an inner axis
 bool isInnerResharding(Expr* expr);
 
+// Shards all tensors in tvs like reference
 void shardAllLike(TensorView* ref, std::vector<TensorView*> tvs);
 
+// Shards all tensors within a boundary like ref_tvs, but does not change the
+// sharding of tvs already sharded.
+void shardBetween(
+    std::vector<TensorView*> ref_tvs,
```

Comment:
The behavior that it only picks `ref_tvs[0]` as the reference tensor is worth calling out. (I understand other tensors are used as the starting points of the traversal). 

Alternatively, we can make it `shardBetween(starts, ends, ref_tv)`. It's slightly more typing but less surprising.

Reply:
I wanted to add a check that all the tensors in ref_tv are sharded consistently.  

I think IdModel could be used for this? We can check if a there is a DID iterdomain then they should all be mapped.